pxr/usdImaging/lib/usdImagingGL/drawModeAdapter.cpp

```diff
@@ -1092,11 +1092,23 @@ UsdImagingGLDrawModeAdapter::_ComputeExtent(UsdPrim const& prim) const
 {
     HD_TRACE_FUNCTION();
     HF_MALLOC_TAG_FUNCTION();
-
-    TfTokenVector purposes = { UsdGeomTokens->default_, UsdGeomTokens->proxy,
-                               UsdGeomTokens->render };
-    UsdGeomBBoxCache bboxCache(UsdTimeCode::EarliestTime(), purposes, true);
-    return bboxCache.ComputeUntransformedBound(prim).ComputeAlignedBox();
+    if (prim.IsLoaded()) {
+        TfTokenVector purposes = { UsdGeomTokens->default_,
+            UsdGeomTokens->proxy, UsdGeomTokens->render };
+        UsdGeomBBoxCache bboxCache(
+            UsdTimeCode::EarliestTime(), purposes, true);
+        return bboxCache.ComputeUntransformedBound(prim).ComputeAlignedBox();
+    } else {
+        GfRange3d extent;
+        if (auto attr = UsdGeomModelAPI(prim).GetExtentsHintAttr()) {
+            VtVec3fArray extentHints;
+            if (attr.Get(&extentHints, UsdTimeCode::EarliestTime()) &&
+                extentHints.size() == 2) {
+                extent = GfRange3d(extentHints[0], extentHints[1]);
```

Comment:
This is incorrect - or rather, it works only for very simple models.  Please reread the documentation for GetExtentsHint(), as it can be an array of up to 8 elements.  In the clause above you are looking for proxy and render purposes in addition to default, so I would recommend that you merge those three extents together here, if present.  If you wanted to add a helper UsdGeomModelAPI::ComputeExtentsHintForPurposes() that _would_ return a simple extent, I wouldn't object, though I also wouldn't require it.